torchaudio/transforms.py

```diff
@@ -52,6 +52,12 @@ class Spectrogram(torch.nn.Module):
             :attr:`center` is ``True``. Default: ``"reflect"``
         onesided (bool, optional): controls whether to return half of results to
             avoid redundancy Default: ``True``
+        return_complex (bool, optional):
+            ``return_complex = True``, this function returns the resulting Tensor in
+            complex dtype, otherwise it returns the resulting Tensor in real dtype with extra
+            dimension for real and imaginary parts. (see ``torch.view_as_real``).
+            When ``power`` is provided, the value must be False, as the resulting
+            Tensor represents real-valued power.
```

Comment:
I'd suggest adding a warning that `Spectrogram now supports native complex tensors. Currently we return pseudo complex tensors (..., 2) by default, but that will change in the future and `return_complex` would be set to `True` by default` something along those lines

Reply:
I like the idea but in domain libraries, it is agreed to not mention the migration plan or future plan in docstring. (@cpuhrsch brought this up and this is meant to follow the same approach as PyTorch. But if PyTorch also mentions future plan in docstring, please point us to it.)

The general idea behind of it is, this kind of direction should better live in release note or issue.

Reply:
We try to always mention the new path.  To me, the definition of deprecation is two parts:
1.  this isn't recommended anymore
2. this is what you should do instead

you could, I guess, but those in the release notes by why make it more difficult for your users?

Reply:
Oh sorry I misread the comment by @anjali411 as adding the suggested sentence to `docstring`. Yes, there is nothing goes against on adding it to warning.

Reply:
I will make a follow up PR.